### ISSUE TITLE
LlaMa-VID: An Image is Worth 2 Tokens in LLMs

### ISSUE THREAD

**Issue by amyeroberts** [huggingface/transformers#27980]
### Model description

LLaMA-VID is an open-source chatbot trained by fine-tuning LLaMA/Vicuna on GPT-generated multimodal instruction-following data. LLaMA-VID empowers existing frameworks to support hour-long videos and pushes their upper limit with an extra context token. We build this repo based on LLaVA.

LLaMA-VID contains three parts: encoder and decoder are adopted to produce visual embedding and text-guided features, respectively; context token and content token are transformed with the tailored token generation strategy; instruction tuning is designed to unleash the potential of LLMs for image and video.

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

Page: https://llama-vid.github.io/
Weights already available on HF: https://huggingface.co/YanweiLi/llama-vid-7b-pretrain-224
Code: https://github.com/dvlab-research/LLaMA-VID

**Comment by venkateshtata** [comment]
Hi @amyeroberts , Can I take up the task of porting this model  ?

**Comment by amyeroberts** [comment]
@venkateshtata Certainly! Feel free to open a PR and ping me when it's ready for review. Let us know if you have any questions about porting into transformers in the meantime :)

**Comment by nileshkokane01** [comment]
@amyeroberts ,

Its been couple of months there is no activity.
Can I give a try ?

**Comment by amyeroberts** [comment]
Sure! We prioritise reviewing based on when PRs are open, rather than comments on issues, as we find this prevents them from becoming too stale. You're free to open a PR to address any issue with no linked PR, or with a PR which is old and hasn't been reviewed / isn't active

**Comment by amyeroberts** [comment]
@nileshkokane01 I wouldn't say responsible - after all they're not asking us to add the model! 

Who to ask depends on what the issue is. If you're unable to replicate numbers just following the instructions on the [model's README](https://github.com/dvlab-research/LLaMA-VID?tab=readme-ov-file#install), then I would reach out to them, probably by opening an issue on the repo first. 

If however it's an issues when porting the original model to transformers, it's best to ask us on a draft PR of the port so we can see the code.

**Comment by nileshkokane01** [comment]
@amyeroberts ,
Thanks, that helps .

**Comment by qgallouedec** [comment]
Since this was opened, our policy for new architectures has settled on the "Model on the Hub first" path: new models land as remote-code repos loadable via `trust_remote_code=True` rather than native core ports by default.
LLaMA-VID is a LLaVA-based video LLM, and that family is now well covered natively (e.g. `LlavaNextVideo`, `LlavaOnevision`, `VideoLlava`, `VideoLlama3`), so with no PR having materialized in the ~2.5 years since, we're going to close this as won't-implement.
